### PR TITLE
 fix: Fixes #288. Alphabetical ordering for Sections and Methods of RPC, Storage, and Extrinsics

### LINKED ISSUE #288: Sections and Methods on Node Interface menu pages of RPC, Storage, and Extrinsics are not in ascending alphabetical order
The ordering of sections and the ordering of methods listed in Node Interface menu pages RPC, Storage, and Extrinsics are not in alphabetical order.

Refer to screenshot below. Sections are not in alphabetical order since "consensus" appears before "balances". Method are not in alphabetical order since "reportMisbehavior" appears before "noteOffline"

screen shot 2018-10-14 at 23 41 35

### PR DESCRIPTION
**Pull request by ltfschoen**
* Ascending alphabetical ordering of Sections and Methods for each of the Node Interface pages including RPC, Storage and Extrinsics

* The only exception is the "substrate" Section on the Storage page. Whilst Methods in the "substrate" Section have been manually ordered in alphabetical. The "substrate" Section itself is appended at the end since it is loaded from a separate file, as explained in Issue https://github.com/polkadot-js/api/issues/290 and will be dealt with in a separate PR

### CODE REVIEW

File: packages/types/src/scripts/MetadataMd.ts [hunk 1]
```diff
@@ -40,18 +42,40 @@ function addRpc () {
   }, renderHeading);
 }
 
+function sortExtrinsicsSectionMethods () {
+  return function (a: any, b: any) {
+    const nameA = a.raw.name.raw.toUpperCase(); // ignore upper and lowercase
+    const nameB = b.raw.name.raw.toUpperCase(); // ignore upper and lowercase
+
+    if (nameA < nameB) {
```
Comment (jacogr) [inline packages/types/src/scripts/MetadataMd.ts]: Here we should just be able to do `nameA.localaCompare(nameB)` (conveted to strings above)

**Comment by ltfschoen** [comment]
@jacogr I've pushed a commit addressing the review comment

**Comment by jacogr** [comment]
Just dropping it here - we need to do the same for Events (even through the API currently doesn not expose it, it should - we can get the docs in place there)

**Comment by ltfschoen** [comment]
@jacogr I've resolved the merge conflicts

**Pull request merged**